### PR TITLE
initial commit of Get-DbaProcess

### PR DESCRIPTION
**Pull request by ctrlbold**
Stop-DbaProcess does not support the output of this :| I need to research how to properly do that. My way feels hacky.

### CODE REVIEW

**Comment by ctrlbold** [comment]
- [ ] Get-DbaProcess -SqlServer localhost -Programs 
- [ ] Get-DbaProcess -SqlServer localhost 
- [] Get-DbaProcess -SqlServer localhost -Logins "$env:USERDOMAIN\$env:USERNAME", sa
- [] $credential = Get-Credential; Get-DbaProcess -SqlServer localhost -SqlCredential $credential 
- [] Get-DbaProcess -SqlServer sql2014 -Exclude 1
- []  Get-DbaProcess -SqlServer sql2014 -Databases msdb -Logins "$env:USERDOMAIN\$env:USERNAME"

1. First example, type that in then tab through to see what programs populate. Find one then press enter Get-DbaProcess -SqlServer sql2014 -Programs 'dbatools PowerShell module - dbatools.io'
2. Should show all processes
3. Should show your own processes and sa
4. Pass a valid credential. SQL Login or Windows Auth should work.
5. Should show all processes except for the one with a spid of 1
6. Should show all processes for msdb AND your own login

I can already see a better way to do excludes. Will include that in November. Like ExcludeLogins, ExcludeDatabases. This will not work if you pipe it to Stop-DbaProcess. This is a known issue.

File: functions/Get-DbaProcess.ps1 [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+		}
+	}
+	
+	END
+	{
+		$sourceserver.ConnectionContext.Disconnect()
+		
+		if ($Detailed)
+		{
+			return $allsessions | Select-Object Spid, Login, Host, Status, Command, Database, Cpu, MemUsage, BlockingSpid, IsSystem, Program
```
Comment (ClaudioESSilva) [inline functions/Get-DbaProcess.ps1]: EnumPorcesses() don't display info on `Status `and `Command` when query is sleeping. 
On sp_who2 this shows `'sleeping'` and `'AWAITING COMMAND'`

image

```
if ($Detailed)
		{
			return $allsessions | Select-Object Spid, Login, Host, Database, BlockingSpid, Program, @{ name = "Status"; expression = {if ($_.Status -eq "") {"sleeping"} else {$_.Status}} },  @{ name = "Command"; expression = {if ($_.Command -eq "") {"AWAITING COMMAND"} else {$_.Command}} }, Cpu, MemUsage, IsSystem
		}

		return ($allsessions | Select-Object Spid, Login, Host, Database, BlockingSpid, Program, @{ name = "Status"; expression = {if ($_.Status -eq "") {"sleeping"} else {$_.Status}} },  @{ name = "Command"; expression = {if ($_.Command -eq "") {"AWAITING COMMAND"} else {$_.Command}} })
```
This way, we keep it 100% SMO :smile: 

I also have changed columns' order (OCD strikes! :see_no_evil: )

File: functions/Get-DbaProcess.ps1 [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+		{
+			$allsessions = $processes
+		}
+		
+		if ($exclude.count -gt 0)
+		{
+			$allsessions = $allsessions | Where-Object { $exclude -notcontains $_.Spid }
+		}
+		
+		$duplicates = @()
```
Comment (ClaudioESSilva) [inline functions/Get-DbaProcess.ps1]: From this line (141) to 154 the code can be removed.

File: functions/Get-DbaProcess.ps1 [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+		}
+		
+		if ($logins.count -gt 0)
+		{
+			$allsessions += $processes | Where-Object { $_.Login -in $Logins }
+		}
+		
+		if ($spids.count -gt 0)
+		{
+			$allsessions += $processes | Where-Object { $_.Spid -in $spids }
```
Comment (ClaudioESSilva) [inline functions/Get-DbaProcess.ps1]: Add `-or $_.BlockingSpid -in $spids`
This way we can know which spids are being blocked by the spid we are searching.

File: functions/Get-DbaProcess.ps1 [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+This command displays processes associated with a spid, login, host, program or database.
+
+.PARAMETER SqlServer
+The SQL Server instance.
+
+.PARAMETER SqlCredential
+Allows you to login to servers using SQL Logins as opposed to Windows Auth/Integrated/Trusted. 
+
+.PARAMETER Spids
+This parameter is auto-populated from -SqlServer. You can specify one or more Spids to be displayed.
```
Comment (ClaudioESSilva) [inline functions/Get-DbaProcess.ps1]: Add something like "Also search on BlockingSpid column".
This is related with line 113

**Pull request merged**